power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -163,38 +178,506 @@ template <bool sym> class NewtonRaphsonSESolver {
+
+                // get a reference and reset block to zero
+                NRSEGainBlock<sym>& block = data_gain_[data_idx_lu];
+                if (row == col) {
+                    // fill block with voltage measurement, only diagonal
+                    process_voltage_measurements(diag_block, rhs_block, measured_values, row);
+                } else {
+                    // Diagonal block is being cleared outside this loop
+                    block.clear();
+                }
+
+                // fill block with branch, shunt measurement
+                for (Idx element_idx = y_bus.y_bus_entry_indptr()[data_idx];
+                     element_idx != y_bus.y_bus_entry_indptr()[data_idx + 1]; ++element_idx) {
+                    Idx const obj = y_bus.y_bus_element()[element_idx].idx;
+                    YBusElementType const type = y_bus.y_bus_element()[element_idx].element_type;
+
+                    switch (type) {
+                    case YBusElementType::shunt:
+                        if (measured_values.has_shunt(obj)) {
+                            auto const& ys = param.shunt_param[obj];
+                            auto const& measured_power = measured_values.shunt_power(obj);
+                            process_shunt_measurement(block, rhs_block, ys, u_state, measured_power);
+                        }
+                        break;
+                    case YBusElementType::bft:
+                        [[fallthrough]];
+                    case YBusElementType::btf: {
+                        auto const& y_branch = param.branch_param[obj];
+                        if (measured_values.has_branch_from(obj)) {
+                            auto const ij_voltage_order = (type == YBusElementType::bft);
+                            process_branch_measurement(block, diag_block, rhs_block, y_branch.yff(), y_branch.yft(),
+                                                       u_state, ij_voltage_order,
+                                                       measured_values.branch_from_power(obj));
+                        }
+                        if (measured_values.has_branch_to(obj)) {
+                            auto const ij_voltage_order = (type == YBusElementType::btf);
+                            process_branch_measurement(block, diag_block, rhs_block, y_branch.ytt(), y_branch.ytf(),
+                                                       u_state, ij_voltage_order, measured_values.branch_to_power(obj));
+                        }
+                        break;
+                    }
+                    default:
+                        assert(type == YBusElementType::bff || type == YBusElementType::btt);
+                        break;
+                    }
+                }
+
+                // fill block with injection measurement constraints
+                if (measured_values.has_bus_injection(row)) {
+                    auto const& yij = y_bus.admittance()[data_idx];
```

Comment:
This line accesses the element `(i,j)` in the y_bus admittance matrix, is `data_idx` then the correct index to use here?

Reply:
yes cause `data_idx` by design is the data element that accesses the sparse matrix `y_bus.admittance()` in the correct data entry to obtain the element representing the element at `(i,j)`

Reply:
Awesomo.